### PR TITLE
remove stylelint dependency

### LINKED ISSUE #2053: Missing dependency on node_modules reinstall
# 🐛 Bug Report

### 😯 Current Behavior

I ran the following sequence:

```
npx create-yoshi-app
yarn yoshi build
rm -rf node_modules
yarn
yarn yoshi build
```
Last step failed with

```
// Type error: Could not find a declaration file for module 'vfile-message'. '/tmp/a/node_modules/vfile-message/index.js' implicitly has an 'any' type.
```

### 🤔 Expected Behavior

Reinstalling removed dependencies should be enough to continue building project

### 💁 Possible Solution 

To fix, I had to add missing definition to my project's `devDependencies`:

```
yarn add --dev @types/vfile-message
```

### 🌍 Your Environment

```
  System:
    OS: Linux 4.15 Ubuntu 18.04.4 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz
  Binaries:
    Node: 12.13.1 - ~/.nvm/versions/node/v12.13.1/bin/node
    Yarn: 1.21.1 - ~/.nvm/versions/node/v12.13.1/bin/yarn
    npm: 6.12.1 - ~/.nvm/versions/node/v12.13.1/bin/npm
    Watchman: Not Found
  Browsers:
    Chrome: 80.0.3987.122
    Firefox: 73.0.1
  npmPackages:
    @storybook/core: Not Found
    @storybook/react: Not Found
    babel-preset-yoshi:  4.37.1 
    eslint-config-yoshi:  4.44.0 
    eslint-config-yoshi-base:  4.44.0 
    jest-yoshi-preset:  4.44.0 
    storybook: Not Found
    tslint-config-yoshi:  4.32.0 
    tslint-config-yoshi-base:  4.22.0 
    typescript: ^3.7.5 => 3.7.5 
    webpack:  4.42.0 
    yoshi-config:  4.43.2 
    yoshi-helpers:  4.43.2 
    yoshi-runtime:  1.0.701 

 Zero config 🛴

```

### PR DESCRIPTION
**Pull request by ranyitz**
### 🔦 Summary

**TL;DR - Yoshi will stop bringing stylelint as a dependency, instead `stylelint-config-yoshi` will bring the relevant stylelint dependencies so users that use it will not break**

___________

Yoshi brings stylelint as part of its dependencies, it means that every Yoshi user gets it, even if they aren't using it. After [searching through Wix's codebase](https://github.com/search?p=1&q=org%3Awix-private+stylelint-config-yoshi&type=Code) It doesn't looks like it has been used broadly.

It is quite heavy package and it would be nice to download less code if possible.

https://bundlephobia.com/result?p=stylelint@9.8.0
image

The proposed solution is to remove all stylelint related dependencies. In order to use stylelint, users need to specify `stylelint-config-yoshi` and they will get `stylelint` and `stylelint-prettier`.

It will be a breaking change though, for people that doesn't specify `stylelint-config-yoshi` in their `package.json`'s dependencies. (less than 5 occurrences)

It will also Fix #2053

### CODE REVIEW

**Comment by romansh-wixcom** [comment]
Hey! 
Not sure if it's a right place to have a discussion, but still...

Looks like `stylelint-config-yoshi` do not use `stylelint` and just describe a config. So I suggest to move `stylelint` to `peerDependencies` as well. @ranyitz  WDYT?

**Comment by ranyitz** [comment]
@romansh-wixcom That would be a breaking change. Yoshi used to bring `stylelint` in a specific version, See the description above.
